src/analysis/seminorm.lean

```diff
@@ -584,6 +590,61 @@ lemma comp_smul_apply (p : seminorm 𝕜 F) (f : E →ₗ[𝕜] F) (c : 𝕜) (x
 
 end semi_normed_comm_ring
 
+section normed_field
+variables [normed_field 𝕜] [add_comm_group E] [module 𝕜 E]
+
+private lemma bdd_below_range_add (x : E) (p q : seminorm 𝕜 E) :
+  bdd_below (range (λ (u : E), p u + q (x - u))) :=
+by { use 0, rintro _ ⟨x, rfl⟩, exact add_nonneg (p.nonneg _) (q.nonneg _) }
+
+noncomputable instance : has_inf (seminorm 𝕜 E) :=
+{ inf := λ p q,
+  { to_fun := λ x, ⨅ u : E, p u + q (x-u),
+    triangle' := λ x y, begin
+      rw ← sub_le_iff_le_add,
+      refine le_cinfi (λ u, _),
+      rw sub_le,
+      refine le_cinfi (λ v, _),
+      rw sub_le_iff_le_add,
+      apply cinfi_le_of_le (bdd_below_range_add _ _ _) (v+u), dsimp only,
```

Comment:
This is pretty painful and the reasoning could be useful elsewhere, why not adding:
```lean
lemma cinfi_le_cinfi_add_cinfi {α : Type*} {ι : Sort*} [conditionally_complete_lattice α]
   [add_comm_group α] [covariant_class α α (+) (≤)] [covariant_class α α (function.swap (+)) (≤)]
  [nonempty ι] {f g h : ι → α} (H : ∀ i j, infi f ≤ g i + h j) : infi f ≤ infi g + infi h :=
(add_comm (infi h) (infi g)) ▸ 
  sub_le_iff_le_add.mp (le_cinfi (λ u, sub_le.mp (le_cinfi (λ v, sub_le_iff_le_add.mpr (H v u)))))
```
somewhere?

Reply:
Looks like something that should be done on another PR

Reply:
No.

Reply:
As you added it, I marked the conversation as resolved